File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.java
class Solution {
    public int maxProfit(int[] prices) {
        int p1 = 0;
        int p2 = 1;
        int bestP = 0;
        while (p2 < prices.length){
            int res = prices[p2]-prices[p1];
            if (res > bestP){
                bestP = res;
            } else if (res<0){
                p1 = p2;
            }
            p2++;
        }
        return bestP;
    }
}

File: Data Structures & Algorithms/buy-and-sell-crypto-with-cooldown/submission-0.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        def recur(i,coin):
            if i >= len(prices):
                return 0
            r1 = 0
            if not coin:
                r1 = max(recur(i+1,1) - prices[i],r1)
            else:
                r1 = max(recur(i+2,0) + prices[i],r1)
            
            r1 = max(recur(i+1,coin), r1)

            return r1
        
        return recur(0,0)